libs/dal/src/repositories/notification-template/notification-template.repository.ts

```diff
@@ -47,27 +47,44 @@ export class NotificationTemplateRepository extends BaseRepository<
     return this.mapEntity(item);
   }
 
-  async findBlueprint(idOrIdentifier: string) {
-    if (!this.blueprintOrganizationId) throw new DalException('Blueprint environment id was not found');
-
+  buildIdOrIdentifierQuery(idOrIdentifier: string) {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** This seems like something that may be handled better at the service layer rather than making than opening up this query to multiple input types. While this may be a bit of an exaggeration, if we continue with this pattern, we could end up with a single query for 7 different identifier types.

I'd love to hear other opinions on it, but I think this query should be "closed" for modification and instead create a separate query function for the other type of identifier. This allows another function to delegate which query should be used. I also imagine a worst-case scenario that could involve `|` across an unindexed field resulting in terrible performance for every execution.

Definitely not a blocker, but perhaps a discussion item!‏

Reply:
Firstly, let me say that I completely agree. I really dislike having to add it here instead of at the application layer. Now, let me explain why it's included here, and if you have any other suggestions, I'd be more than happy to hear them, given the reasons mentioned above.

Here's why it's included here:

- We want to provide users of those endpoints, starting with the blueprint and now the template itself, the ability to fetch by the identifier in cases where our user might not be aware of the `_id`.
- The main reason is the validation we perform on the query payload build. To construct a proper payload, we need to determine whether it is a MongoDB ID or not.

Reply:
But you got me thinking 👀, let me try something out.

Reply:
Done! :)
let me know what you think, and btw thanks for questioning the implementation ♥️